### PR TITLE
Import into: add job management statement

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #42930 

Problem Summary:

### What is changed and how it works?
- add system table `mysql.tidb_import_jobs`
- redact secret keys when storing it
- add show import job(s) statement
    - it shows created by current user(no need privileges to system table), or if user have SUPER privileges, it can see all jobs
    - for running jobs, we get imported_rows from distributed framework
- add cancel import job statement
    - cancel job created by current user(no need privileges to system table), or if user have SUPER privileges, it can cancel any job
- Unify Return result of IMPORT INTO either run in detached mode or not

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/storage/task_table.go [hunk 1]
```diff
@@ -158,29 +159,35 @@ func (stm *TaskManager) executeSQLWithNewSession(ctx context.Context, sql string
...
+	}
 
-		return nil
-	})
+	rs, err := execSQL(stm.ctx, se, "select @@last_insert_id")
+	if err != nil {
+		return 0, err
+	}
 
+	taskID, err = strconv.ParseInt(rs[0].GetString(0), 10, 64)
```
Comment (lance6716) [inline disttask/framework/storage/task_table.go:184]: found a bug https://github.com/pingcap/tidb/issues/44574

**Review by lance6716 — COMMENTED** [review]
13 / 39 files viewed

File: executor/importer/import.go [hunk 2]
```diff
@@ -187,6 +188,11 @@ type Plan struct {
 
 	// todo: remove it when load data code is reverted.
 	InImportInto bool
+	// only initialized for IMPORT INTO, used when creating job.
+	Parameters *ImportParameters `json:"-"`
+	// the user who executes the statement,in the form of user@host
```
Comment (GMHDBJD) [inline executor/importer/import.go:193]:
```suggestion
	// the user who executes the statement, in the form of user@host
```

File: executor/importer/job.go [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+// GetJob returns the job with the given id if the user has privilege.
+// hasSuperPriv: whether the user has super privilege.
+// If the user has super privilege, the user can show or operate all jobs,
+// else the user can only show or operate his own jobs.
+func GetJob(ctx context.Context, conn sqlexec.SQLExecutor, jobID int64, user string, hasSuperPriv bool) (*JobInfo, error) {
+	ctx = util.WithInternalSourceType(ctx, kv.InternalImportInto)
+
+	sql := baseQuerySQL + ` WHERE id = %?`
+	args := []interface{}{jobID}
+	rs, err := conn.ExecuteInternal(ctx, sql, args...)
```
Comment (lance6716) [inline executor/importer/job.go:129]: we can directly use `jobID`?

File: disttask/framework/storage/task_table.go (hunk 1 above)
Comment (D3Hunter) [reply disttask/framework/storage/task_table.go:184]: seems not related to this pr?

Comment (lance6716) [reply disttask/framework/storage/task_table.go:184]: yes

File: executor/importer/job.go [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+var (
+	// TestLastImportJobID last created job id, used in unit test.
+	TestLastImportJobID atomic.Int64
+)
+
+// constants for job status and step.
+const (
+	// JobStatus
+	// 	┌───────┐     ┌───────┐    ┌────────┐
+	// 	│pending├────►│running├───►│finished│
```
Comment (GMHDBJD) [inline executor/importer/job.go:44]: pending -> running -> reverting -> reverted/revert_failed

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

**Review by lance6716 — APPROVED** [review]

File: executor/importer/job.go (hunk 4 above)
Comment (D3Hunter) [reply executor/importer/job.go:44]: that's frameworks's state transition, we don't have revert for job

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: parser/parser.y [hunk 5]
```diff
@@ -1034,7 +1034,7 @@ import (
 	ShardableStmt              "Shardable statement that can be used in non-transactional DMLs"
 	PauseLoadDataStmt          "PAUSE LOAD DATA JOB statement"
 	ResumeLoadDataStmt         "RESUME LOAD DATA JOB statement"
-	CancelLoadDataStmt         "CANCEL LOAD DATA JOB statement"
+	CancelImportStmt           "CANCEL LOAD DATA JOB statement"
```
Comment (bb7133) [inline parser/parser.y:1037]: Please update the comment.

File: br/pkg/storage/parse.go [hunk 6]
```diff
@@ -207,6 +208,31 @@ func FormatBackendURL(backend *backuppb.StorageBackend) (u url.URL) {
...
+func RedactURL(str string) (string, error) {
+	u, err := ParseRawURL(str)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	scheme := u.Scheme
+	failpoint.Inject("forceRedactURL", func() {
+		scheme = "s3"
+	})
+	if strings.ToLower(scheme) == "s3" {
```
Comment (bb7133) [inline br/pkg/storage/parse.go:221]: How about defining some constants for "s3", "access-key", "secret-access-key"?

Comment (D3Hunter) [reply br/pkg/storage/parse.go:221]: there're many such string literal in this package, will not do it in this pr

Comment (bb7133) [reply br/pkg/storage/parse.go:221]: Sure~

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**